### PR TITLE
CASM-4653 Implementation of a method to collect and print the "onExit" hook podlogs

### PR DESCRIPTION
**Pull request by Pankhuri-Rajesh**
## Summary and Scope

The code changes are related to the issue of the podlogs not being tracked for onExit hook scripts by iuf-cli. So, a separate method to find the "onExit" hook pod and collect the logs is implemented.  
The onExitHandlers were added in the backend https://github.com/Cray-HPE/cray-nls/pull/206 . This cli change is made accordingly.
JIRA- https://jira-pro.it.hpe.com:8443/browse/CASM-4653
### Tested on:

  * Virtual Shasta - beau

### Test description:

Code changes were tested by adding dummy onExit hook script in pre-install-check for COS .

## Pull Request Checklist

- [ ] Version number(s) incremented, if applicable
- [ ] Copyrights updated
- [ ] License file intact
- [x] Target branch correct
- [ ] CHANGELOG.md updated
- [x] Testing is appropriate and complete, if applicable
- [ ] [HPC Product Announcement](https://cray.slack.com/archives/C026TVCSXLH) prepared, if applicable

### CODE REVIEW

**Review by leliasen-hpe — APPROVED** [review]

**Review by atifsyedali — APPROVED** [review]

**Review by larry-tordsen — APPROVED** [review]

File: lib/Activity.py [hunk 1]
```diff
@@ -621,7 +622,7 @@ def monitor_workflow(self, workflow):
 
             try:
                 completed = wflow['metadata']['labels']['workflows.argoproj.io/completed']
-                if completed and completed == 'true':
+                if completed and completed == 'true':
```
Comment (JohnNavitskyHPE) [inline lib/Activity.py]: trailing whitespace?

File: lib/Activity.py [hunk 2]
```diff
@@ -527,7 +527,7 @@ def dfs(data, start):
                 slist[name] = node["children"]
             else:
                 slist[name] = []
-
+
```
Comment (JohnNavitskyHPE) [inline lib/Activity.py]: extra whitespace?

File: lib/Activity.py [hunk 3]
```diff
@@ -648,14 +649,45 @@ def monitor_workflow(self, workflow):
...
+                                    for container in ["init", "wait", "main"]:
+                                        proc = multiprocessing.Process(target=self.podlogs.follow_pod_log, args=(podname, container, log_prefix, self.st_event))
+                                        proc.start()
+                                        self.running_procs.append(proc)
+                                except:
+                                    pass
 
             npath = self.sort_phases(workflow, nodes)
-
+
```
Comment (JohnNavitskyHPE) [inline lib/Activity.py]: whitespace

**Review by JohnNavitskyHPE — APPROVED** [review]

File: lib/Activity.py (hunk 3 above)
Comment (Pankhuri-Rajesh) [reply lib/Activity.py]: removed all whitespaces

**Review by rajeshranganathan-hpe — APPROVED** [review]

**Pull request merged**